apps/engine-test/src/orbitControls.js

```diff
@@ -0,0 +1,235 @@
+/** Based on orbitControls from @openjscad/regl-renderer 
+ and https://webglfundamentals.org/webgl/lessons/webgl-3d-camera.html
+ and
+   https://elalish.blogspot.com/2022/04/3d-interaction.html
+   https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/three-components/SmoothControls.ts
+*/
+import * as mat4 from 'gl-matrix/esm/mat4.js'
+import * as vec3 from 'gl-matrix/esm/vec3.js'
+
+const { hypot, acos, asin, PI, cos, sin } = Math
+const PId2 = PI / 2
+const TAU = PI * 2
+
+export class OrbitCamera {
+  position = [180, -180, 220]
+  target = [0, 0, 0]
+}
+
+export class OrbitState {
+  position
+  target
+  calc
+  constructor(position = [180, -180, 220], target = [0, 0, 0], ...args) {
+    if (args.length) {
+      // clone
+      this.position = position
+      this.target = target
+      this.calc = args[0]
+    } else {
+      this.set(position, target, false)
+    }
+  }
+
+  set(position, target, fire = true) {
+    if (target) this.target = target
+    if (position) this.position = position
```

Comment:
The conditional shouldn't be necessary as it already has default values.

Reply:
conditional is for when you want to change only one of them. There is no default value for the param itself for that same reason.